File: windows/Scanner/Rules/AESMCP010.cs
using AIExposureScanner.Scanner.Models;

namespace AIExposureScanner.Scanner.Rules;

public sealed class AESMCP010 : IRule
{
    public string Id => "AES-MCP-010";
    public Severity Severity => Severity.High;

    public IReadOnlyList<Finding> Evaluate(ScanFacts facts) =>
        facts.McpServers
            .Where(server => !server.Disabled)
            .Where(server => server.SearchableValues().Any(AiThreatPatterns.IsPlaintextHttpEndpoint))
            .Select(server => this.Finding(server.AppId, server.Name, affectedPath: server.ConfigPath))
            .ToArray();
}

File: macos/Scanner/Sources/Scanner/Rules/AESMCP010.swift
public struct AESMCP010: Rule {
    public let id = "AES-MCP-010"
    public let severity: Severity = .high

    public init() {}

    public func evaluate(_ facts: ScanFacts) -> [Finding] {
        facts.mcpServers.compactMap { server in
            guard !server.disabled else {
                return nil
            }
            let usesPlaintextHttp = server.searchableValues.contains { value in
                AiThreatPatterns.isPlaintextHttpEndpoint(value)
            }
            guard usesPlaintextHttp else {
                return nil
            }
            return finding(app: server.appId, serverName: server.name, affectedPath: server.configPath)
        }
    }
}
